(MAINT) Fix dujour issues

fpringvaldsen opened the pull request:
Fix an issue wherein the group-id "puppetlabs" was not being used,
which caused getting the version string for puppet-server to fail.
Make dujour configuration optional. Remove dujour configuration
from the default config file.

camlow325 commented on src/clj/puppetlabs/services/master/master_core.clj:
```diff
@@ -69,14 +69,14 @@
-       "  Puppet Server version: " (version-check/get-version-string "puppet-server") "\n"
+       "  Puppet Server version: " (version-check/get-version-string (:artifact-id product-name) (:group-id product-name)) "\n"
        "  Puppet version: " (:puppet-version (config/get-puppet-config jruby-service)) "\n"
        "  Supported /puppet API versions: " puppet-API-versions
```
Might be nice to add a "\n" at the end of this line so that it doesn't run into the next one?

camlow325 commented on src/clj/puppetlabs/services/master/master_service.clj:
```diff
@@ -23,9 +23,9 @@
-         update-server-url (get-in config [:product :update-server-url])]
+         product-name      (or (get-in config [:product :name]) {:group-id "puppetlabs" :artifact-id "puppet-server"})
+         upgrade-error     (core/construct-404-error-message jruby-service product-name)
+         update-server-url (or (get-in config [:product :update-server-url]) "http://updates.puppetlabs.com")]
```
Do we want/need to code the "http://updates.puppetlabs.com" default in here?  If it is passed through as nil, doesn't the dujour-version-check library use "http://updates.puppetlabs.com" by default?  Wasn't sure we wanted to have two levels of defaults - in puppet-server and the underlying library - for the case that it doesn't appear in the config file.

camlow325 commented:
:see_no_evil: with just the questions about possibility adding a newline to the 404 error message and whether we need to duplicate the "http://updates.puppetlabs.com" default in puppet-server.

@jeffmccune - you probably want to take a look at this assuming we're going to try to get this in for the rc build.

cprice404 commented:
+1 but I agree with @camlow325 's comment about not having the URL hard-coded in two places.

fpringvaldsen commented:
@camlow325 @cprice404 I've addressed feedback. Also, ran the ca-files-test without specifying the product information and ensured that dujour isn't logging a connect exception when trying to connect to the default URL, so I think this should be good to go once travis finishes.

camlow325 commented:
:+1: when Travis finishes.

jeffmccune commented:
If this gets merged into master, please update https://tickets.puppetlabs.com/browse/SERVER-457 to that effect.  I know it's a MAINT commit, but I ended up creating a ticket since this is relevant to the RC.

camlow325 commented:
Tired of waiting for Travis and tests pass locally for me, so merging.
